### PR TITLE
Add functional test data for new validation rules + code polish

### PR DESCRIPTION
**Pull request by EasyRhinoMSFT**

### CODE REVIEW

File: src/Sarif.Multitool.Library/Rules/RuleResources.resx [hunk 1]
```diff
@@ -584,4 +584,126 @@ The SARIF standard was developed over several years, and many intermediate versi
   <data name="GH1017_ProvideRequiredPhysicalLocationProperties_Error_MissingArtifactLocationUri_Text" xml:space="preserve">
     <value>{0}: The 'artifactLocation' object on this 'physicalLocation' object does not provide a 'uri' object. This property is required by the {1} service.</value>
   </data>
+  <data name="ADO1011_ReferenceFinalSchema" xml:space="preserve">
```
Comment (EasyRhinoMSFT) [inline src/Sarif.Multitool.Library/Rules/RuleResources.resx:587]: Append '_FullDescription_Text'

**Comment by EasyRhinoMSFT** [comment]
@stacywray is added to the review. #Closed

File: src/Sarif.Multitool.Library/Rules/RuleResources.Designer.cs [hunk 2]
```diff
@@ -78,6 +78,30 @@ internal class RuleResources {
             }
         }
         
+        /// <summary>
+        ///   Looks up a localized string similar to The &apos;$schema&apos; property must refer to the final version of the SARIF 2.1.0 schema. This enables IDEs to provide Intellisense for SARIF log files.
+        ///
```
Comment (stacywray) [inline src/Sarif.Multitool.Library/Rules/RuleResources.Designer.cs:83]: Should "The" be capitalized?

File: src/Sarif.Multitool.Library/Rules/RuleResources.Designer.cs [hunk 3]
```diff
@@ -78,6 +78,30 @@ internal class RuleResources {
...
+        ///The SARIF standard was developed over several years, and many intermediate versions of the schema were produced. Now that the standard is final, only the OASIS standard version of the schema is valid..
+        /// </summary>
```
Comment (stacywray) [inline src/Sarif.Multitool.Library/Rules/RuleResources.Designer.cs:85]: Remove comma before "and"

Comment (stacywray) [inline src/Sarif.Multitool.Library/Rules/RuleResources.Designer.cs:85]: Remove extra period at end of sentence.

File: src/Sarif.Multitool.Library/Rules/RuleResources.Designer.cs [hunk 4]
```diff
@@ -78,6 +78,30 @@ internal class RuleResources {
...
+        internal static string ADO1011_ReferenceFinalSchema {
+            get {
+                return ResourceManager.GetString("ADO1011_ReferenceFinalSchema", resourceCulture);
+            }
+        }
+        
+        /// <summary>
+        ///   Looks up a localized string similar to The root element of a SARIF log file is a SarifLog object. The properties of this element provide information about the log&apos;s schema version as well as an array of analysis runs. These properties are required by the ADO Advanced Security service.
+        ///
```
Comment (stacywray) [inline src/Sarif.Multitool.Library/Rules/RuleResources.Designer.cs:94]: - Uncapitalize "The"

Comment (stacywray) [inline src/Sarif.Multitool.Library/Rules/RuleResources.Designer.cs:94]: - Add ", which" before "is a Sarif object"

File: src/Sarif.Multitool.Library/Rules/RuleResources.resx [hunk 5]
```diff
@@ -584,4 +584,126 @@ The SARIF standard was developed over several years, and many intermediate versi
...
+    <value>The '$schema' property must refer to the final version of the SARIF 2.1.0 schema. This enables IDEs to provide Intellisense for SARIF log files.
+
+The SARIF standard was developed over several years, and many intermediate versions of the schema were produced. Now that the standard is final, only the OASIS standard version of the schema is valid.</value>
+  </data>
```
Comment (stacywray) [inline src/Sarif.Multitool.Library/Rules/RuleResources.resx:591]: Remove comma before "and".

File: src/Sarif.Multitool.Library/Rules/RuleResources.resx [hunk 6]
```diff
@@ -584,4 +584,126 @@ The SARIF standard was developed over several years, and many intermediate versi
...
+  <data name="ADO1013_ProvideRequiredSarifLogProperties_FullDescription_Text" xml:space="preserve">
+    <value>The root element of a SARIF log file is a SarifLog object. The properties of this element provide information about the log's schema version as well as an array of analysis runs. These properties are required by the ADO Advanced Security service.
+
+Provide the '$schema' property, which must refer to the final version of the SARIF 2.1.0 schema. This enables IDEs to provide Intellisense for SARIF log files.
+
+Provide the 'version' property, which must refer to the the final, OASIS standard version of the SARIF 2.1.0 schema. This enables IDEs to provide Intellisense for SARIF log files.
+
+Provide the 'runs' array, even if it is empty. The ADO Advanced Security service specifies a maxzimum of 20 runs.</value>
+  </data>
```
Comment (stacywray) [inline src/Sarif.Multitool.Library/Rules/RuleResources.resx:600]: "Maximum" misspelled - remove "z"

File: src/Sarif.Multitool.Library/Rules/RuleResources.Designer.cs (hunk 2 above)
Comment (EasyRhinoMSFT) [reply src/Sarif.Multitool.Library/Rules/RuleResources.Designer.cs:83]: "Looks up a localized string similar to" is added by VS for use in a tooltip. It's not part of the resource string.

File: src/Sarif.Multitool.Library/Rules/RuleResources.Designer.cs (hunk 3 above)
Comment (EasyRhinoMSFT) [reply src/Sarif.Multitool.Library/Rules/RuleResources.Designer.cs:85]: Second one is added by VS, not part of the string.

File: src/Sarif.Multitool.Library/Rules/RuleResources.resx [hunk 7]
```diff
@@ -584,4 +584,126 @@ The SARIF standard was developed over several years, and many intermediate versi
...
+
+Provide the 'id' property whose value is a stable identifier for the rule.
+
+Provide the 'name' property, which contains a "friendly name" that helps users see at a glance the purpose of the rule. For uniformity of experience across all tools that produce SARIF, the friendly name should be a single Pascal-case identifier, for example, 'ProvideRuleFriendlyName'.</value>
+  </data>
+  <data name="GH1011_ReferenceFinalSchema" xml:space="preserve">
+    <value>The '$schema' property must refer to the final version of the SARIF 2.1.0 schema. This enables IDEs to provide Intellisense for SARIF log files.
+
+The SARIF standard was developed over several years, and many intermediate versions of the schema were produced. Now that the standard is final, only the OASIS standard version of the schema is valid.</value>
+  </data>
```
Comment (stacywray) [inline src/Sarif.Multitool.Library/Rules/RuleResources.resx:652]: Remove comma before "and"

File: src/Sarif.Multitool.Library/Rules/RuleResources.resx [hunk 8]
```diff
@@ -584,4 +584,126 @@ The SARIF standard was developed over several years, and many intermediate versi
...
+  <data name="GH1013_ProvideRequiredSarifLogProperties_FullDescription_Text" xml:space="preserve">
+    <value>The root element of a SARIF log file is a SarifLog object. The properties of this element provide information about the log's schema version as well as an array of analysis runs. These properties are required by the ADO Advanced Security service.
+
+Provide the '$schema' property, which must refer to the final version of the SARIF 2.1.0 schema. This enables IDEs to provide Intellisense for SARIF log files.
+
+Provide the 'version' property, which must refer to the the final, OASIS standard version of the SARIF 2.1.0 schema. This enables IDEs to provide Intellisense for SARIF log files.
+
+Provide the 'runs' array, even if it is empty. The GitHub Advanced Security service specifies a maxzimum of 20 runs.</value>
+  </data>
```
Comment (stacywray) [inline src/Sarif.Multitool.Library/Rules/RuleResources.resx:661]: Remove "z" from "maximum"

File: src/Sarif.Multitool.Library/Rules/RuleResources.resx [hunk 9]
```diff
@@ -584,4 +584,126 @@ The SARIF standard was developed over several years, and many intermediate versi
...
+  <data name="GH2012_ProvideRequiredReportingDescriptorProperties_FullDescription_Text" xml:space="preserve">
+    <value>Rule metadata should provide information that makes it easy to understand and fix the problem.
+
+Provide the 'id' property whose value is a stable identifier for the rule.
+
+Provide the 'shortDescription' and 'fullDescription' properties.
+
+Provide the 'help' property, and populate the 'text' property with information about the rule.</value>
+  </data>
```
Comment (stacywray) [inline src/Sarif.Multitool.Library/Rules/RuleResources.resx:705]: Remove comma before "and"

File: src/Sarif.Multitool.Library/Rules/RuleResources.resx [hunk 10]
```diff
@@ -584,4 +584,126 @@ The SARIF standard was developed over several years, and many intermediate versi
...
+  <data name="Base1011_ReferenceFinalSchema_Error_Default_Text" xml:space="preserve">
+    <value>{0}: The '$schema' property value '{1}' does not refer to the final version of the SARIF 2.1.0 schema. If you are using an earlier version of the SARIF format, consider upgrading your analysis tool to produce the final version. If this file does in fact conform to the final version of the schema, upgrade the tool to populate the '$schema' property with a URL that refers to the final version of the schema.</value>
+  </data>
```
Comment (stacywray) [inline src/Sarif.Multitool.Library/Rules/RuleResources.resx:708]: Add commas around "in fact" or remove "in fact"

**Review by stacywray — APPROVED** [review]
:shipit:

**Pull request merged**